src/routes/GameScreen.js

```diff
@@ -17,25 +17,62 @@ export default class GameScreen extends Component {
         super(props);
 
         this.state = {
-            game: this.props.navigation.state.params.game,
-            canPlay: true,
+            isLoading: true,
+            id: -1,
+            token: '',
+            currentGrid: null,
+            turn: -1,
+            isWinner: null,
         };
+
+        this.requestGame = this.requestGame.bind(this);
+        this.requestMove = this.requestMove.bind(this);
     }
 
     requestGame() {
```

Comment:
```js
requestGame = () => { ... }
```

Reply:
Also instead of using `then`, you could make the `requestGame` function `async` and use `await`.

```js
requestGame = async () => {
    const res = await game()(id, token);

    this.setState({
        isLoading: false,
        id,
        token,
        currentGrid: res.currentPlayer.currentGrid,
        turn: res.currentPlayer.turn,
        isWinner: res.winner !== null,
    });
}
```